src/main/java/processor/Processor.java

```diff
@@ -0,0 +1,182 @@
+            return processBye();
+        case LIST:
+            return processList();
+        case DONE:
+            return processDone(arguments.get(1));
+        case DELETE:
+            return processDelete(arguments.get(1));
+        case FIND:
+            return processFind(arguments.toArray(new String[0]));
+        default:
+            return processDefault(arguments);
+        }
+    }
+
+    /**
+     * Process command implementation ith type DEFAULT.
+     *
+     * @param arguments List of arguments related to the command.
+     * @return Response from the chatbot.
+     */
+    @Override
+    public String processDefault(List<String> arguments) {
+        try {
+            String type = arguments.get(0);
+            if (type.equals("todo")) {
+                arguments.remove(0);
+                if (arguments.size() == 0) {
+                    throw new DukeException("Todo description cannot be empty");
+                }
+                this.storage.addTask(new Todo(String.join(" ", arguments)));
+            } else if (type.equals("deadline")) {
+                arguments.remove(0);
+                if (arguments.size() == 0) {
+                    throw new DukeException("Deadline description cannot be empty");
+                }
+                String line = String.join(" ", arguments);
+                String[] input = line.split(" /by ");
+                if (input.length == 1) {
+                    throw new DukeException("Deadline command must have /by specified");
+                }
+                LocalDate time = LocalDate.parse(input[1].trim());
+                this.storage.addTask(new Deadline(input[0], time));
+            } else if (type.equals("event")) {
+                arguments.remove(0);
+                if (arguments.size() == 0) {
+                    throw new DukeException("Event description cannot be empty");
+                }
+                String line = String.join(" ", arguments);
+                String[] input = line.split(" /at ");
+                if (input.length == 1) {
+                    throw new DukeException("Event command must have /at specified");
+                }
+                LocalDate time = LocalDate.parse(input[1].trim());
+                this.storage.addTask(new Event(input[0], time));
+            } else {
+                throw new DukeException("I don't understand:(");
+            }
+            String response = "Got it. I've added this task:\n   " + this.storage.getLastTask()
+                    + "\nNow you have " + this.storage.getSize() + " tasks in the list.";
+            Ui.print(response);
+            return response;
+        } catch (DukeException e) {
+            Ui.print(e.getMessage());
+            return e.getMessage();
+        } catch (DateTimeParseException e) {
+            Ui.print(e.getMessage());
+            return e.getMessage();
+        }
+    }
+
+    /**
+     * Process command implementation with type LIST.
```

Comment:
I'd prefer "Processes LIST type command" as the comment header. Not sure whether we actually allowed to use nouns.